### PR TITLE
PR: Speeding up Review and Delivery with Auto Sort Imports

### PR DESCRIPTION
**Pull request by kamikazebr**
In the project have some patterns about order of imports, review each commit to check if it's correct its painful.

We can automate it and bring consistency and easy to developer experience, plus fast review and delivery.

The plugin i have tested it's
https://github.com/trivago/prettier-plugin-sort-imports

Using regex to define auto sort like that

image

You can run `yarn prettier` to get correct autosort imports.

### CODE REVIEW

**Comment by rperez89** [comment]
> In the project have some patterns about order of imports, review each commit to check if it's correct its painful.
> 
> We can automate it and bring consistency and easy to developer experience, plus fast review and delivery.
> 
> The plugin i have tested it's https://github.com/trivago/prettier-plugin-sort-imports
> 
> Using regex to define auto sort like that
> 
> image

hey! nice lib! but the import orders are not the correct ones, it would be :
1- React 
2- External libraries + 1hive libs
3- Components
4- Hooks
5- lib + utils + services
6- abis
7- assets

**Comment by kamikazebr** [comment]
> hey! nice lib! but the import orders are not the correct ones, it would be : 1- React 2- External libraries + 1hive libs 3- Components 4- Hooks 5- lib + utils + services 6- abis 7- assets

Nice!!! I'll fix that. But what are `services`? 


> Also wondering why are present on the pr commits from fabri and gabi from other tasks, maybe some merging stuff?

I dont understand the question, what you talk about?

**Comment by kamikazebr** [comment]
@rperez89 reordened, i think you mean `services` but it was `providers`?

image

**Comment by rperez89** [comment]
> @rperez89 reordened, i think you mean `services` but it was `providers`?
> 
> image

Providers goes right after the components, bellow the hooks. Services are the ones under the services folder

**Comment by kamikazebr** [comment]
Now i think i got it right, right?

image

I was not find the @/services because it not was configured webpack alias that way, it just "@" there.

So have other too,  like endpoints and other folders, you all them need order too? @rperez89

**Comment by rperez89** [comment]
> Now i think i got it right, right?
> 
> image
> 
> I was not find the @/services because it not was configured webpack alias that way, it just "@" there.
> 
> So have other too, like endpoints and other folders, do you know order them? Or just leave that way? @rperez89

we can have the endpoints right above the services, let's just try to rebase or something the branch to avoid having the commits from fabri and gabi and just have yours

**Comment by kamikazebr** [comment]
Its weird! Same in the git graph it looks like the gardens-staging

image

I'll check it, maybe cherry-picking on new branch

**Comment by kamikazebr** [comment]
Ahh it was pointed to https://github.com/1Hive/gardens-ui/tree/gardens but i branched from https://github.com/1Hive/gardens-ui/tree/gardens-staging,

File: .prettierrc.json [hunk 1]
```diff
@@ -4,5 +4,23 @@
...
-  "tabSize": 2
-}
\ No newline at end of file
+  "tabSize": 2,
+  "importOrder": [
+    "^react(.*)$",
+    "<THIRD_PARTY_MODULES>",
+    "^@1hive/(.*)$",
+    "^@components/(.*)$",
+    "^@providers/(.*)$",
```
Comment (tmoutinho) [inline .prettierrc.json:13]: The order seems right to me 🤟

**Review by tmoutinho — COMMENTED** [review]
Food for though, why don't we have this as pre-commit hook? Instead of prettier? That would automatically rearange the imports before pushing.What do you guys think?

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
[the remark this answers is not in the document]
Comment (0xGabi) [reply .vscode/launch.json:1]: It was already removed from the new PR.

File: .prettierrc.json [hunk 3]
```diff
@@ -4,5 +4,23 @@
...
+    "^@lib/(.*)$",
+    "^@utils/(.*)$",
+    "^@/services(.*)$",
+    "^@/endpoints(.*)$",
+    "^@/(.*)$",
+    "^@abis/(.*)$",
+    "^@assets/(.*)$",
+    "^[./]"
+  ],
+  "importOrderSeparation": true,
```
[the remark this answers is not in the document]
Comment (0xGabi) [reply .prettierrc.json:24]: Yes, it does. This is how we are currently doing it for the most part. Let's keep it like this and we can consider removing the rule if it generate to much noisse.

**Comment by 0xGabi** [comment]
Hey, @kamikazebr I'll close this one. I applied the new rules with `prettier-ai` as a new single commit that I push directly here: https://github.com/1Hive/gardens-ui/commit/0e03c37e95d3f8bfcdc106f8882491524b017d48

**Comment by 0xGabi** [comment]
Actually, let's wait for #749 before doing the global prettier replacement.

**Pull request closed without merging**